src/store/Appstore.ts

```diff
@@ -5,13 +5,21 @@ import {createJSONStorage, persist} from 'zustand/middleware';
 interface AppState {
   chapterShow: number[];
   setChapterShow: (chapterShow: number[]) => void;
+  toggleChapterStatus: (chapterId: number) => void;
 }
 
 export const useAppStore = createStore<AppState>()(
   persist(
     set => ({
       chapterShow: [],
       setChapterShow: chap => set({chapterShow: chap}),
+      toggleChapterStatus: (chapterId: number) =>
+        set(state => {
+          if (!state.chapterShow.includes(chapterId)) {
```

Comment:
Can I mark my chapter as unread with this implementation?

Reply:
not possible on this version but we can add :
     toggleChapterStatus: (chapterId: number) =>
        set(state => {
          if (!state.chapterShow.includes(chapterId)) {
            return {chapterShow: [...state.chapterShow, chapterId]};
          }
          return {chapterShow: state.chapterShow.filter(id => id !== chapterId)};
        }),
    }),
    
    and delete line 23
    disabled={chapterShow.includes(chapter.id)}